src/cephadm/cephadm

```diff
@@ -1958,24 +1969,33 @@ def infer_config(func: FuncT) -> FuncT:
 
         if 'fsid' in ctx and ctx.fsid:
             name = ctx.name if 'name' in ctx else None
-            if not name:
-                daemon_list = list_daemons(ctx, detail=False)
-                for daemon in daemon_list:
-                    if (
-                        daemon.get('name', '').startswith('mon.')
-                        and daemon.get('fsid', '') == ctx.fsid
-                        and daemon.get('style', '') == 'cephadm:v1'
-                        and os.path.exists(config_path('mon', daemon['name'].split('.', 1)[1]))
-                    ):
-                        name = daemon['name']
-                        break
             if name:
+                # daemon name has been specified, let's use it
                 ctx.config = config_path(name.split('.', 1)[0], name.split('.', 1)[1])
+            else:
+                # no daemon name has been specified, let's see if cluster has a conf dir
+                ceph_conf = f'{ctx.data_dir}/{ctx.fsid}/{CEPH_CONF_DIR}/{CEPH_CONF}'
+                if os.path.exists(ceph_conf):
+                    ctx.config = ceph_conf
+                else:
+                    # finally, if nothing has been provided let's use mon conf file
+                    daemon_list = list_daemons(ctx, detail=False)
+                    for daemon in daemon_list:
+                        if (
+                            daemon.get('name', '').startswith('mon.')
+                            and daemon.get('fsid', '') == ctx.fsid
+                            and daemon.get('style', '') == 'cephadm:v1'
+                            and os.path.exists(config_path('mon', daemon['name'].split('.', 1)[1]))
+                        ):
+                            name = daemon['name']
+                            ctx.config = config_path(name.split('.', 1)[0], name.split('.', 1)[1])
+                            break
```

Comment:
level of indentation here is starting to get uncomfortable. Maybe this should be it's own function?

Reply:
Yeah, I had this feeling also but I was trying to maintain the original logic. I did some refactoring to make the code easier.